### ISSUE TITLE
Accessibility Feedback for Midpoint Review: Decision Tools - Find a VA Form - Add Relevance, Popularity, Related Forms link

### ISSUE THREAD

**Issue by joshkimux** [department-of-veterans-affairs/va.gov-team#26068]
# Accessibility Feedback for Midpoint Review: Decision Tools - Find a VA Form - Add Relevance, Popularity, Related Forms link

## Feedback framework

- **❗️ Must** for if the feedback must be applied 
- **⚠️ Should** if the feedback is best practice 
- **✔️ Consider** for suggestions/enhancements 

## Definition of done

1. Review and acknowledge feedback.
1. Fix and/or document decisions made.
1. Accessibility specialist will close ticket after reviewing documented decisions / validating fix.

## Point of Contact

**VFS Point of Contact:** Josh

<hr/>

## Must
- Helpful links, more information links, and manage your VA benefits links should not rely on color. This is a common issue that can be resolved by referencing past tickets like #18442  
<img width="701" alt="Screen Shot 2021-06-15 at 12 31 23 PM" src="https://user-images.githubusercontent.com/14154792/122090375-b39bea80-cdd5-11eb-9dab-a82213f12632.png">

- Spanish content must have the attribute of `lang="es"` OR be a child of an element with the attribute of `lang="es"` to inherit
<img width="1081" alt="Screen Shot 2021-06-15 at 2 32 23 PM" src="https://user-images.githubusercontent.com/14154792/122105381-b6530b80-cde6-11eb-9a05-53d915f8077b.png">

- Remove the feedback button from the mobile version of the page as it may make filtering completely inaccessible to mobile screen reader users. [A similar problem was flagged during a screen reader pilot test for facilities.](https://dsva.slack.com/archives/C0FQSS30V/p1622755258434100) An epic has been created for [defects related to the feedback button.](https://github.com/department-of-veterans-affairs/va.gov-team/issues/26016)
<img width="378" alt="Screen Shot 2021-06-15 at 3 08 31 PM" src="https://user-images.githubusercontent.com/14154792/122109707-aa1d7d00-cdeb-11eb-8c7f-260afc063e19.png">

## Should
- Avoid using `b` as a replacement for card headings. Use a `h3` instead.
<img width="1063" alt="Screen Shot 2021-06-15 at 3 02 24 PM" src="https://user-images.githubusercontent.com/14154792/122109041-ea303000-cdea-11eb-8463-70492d1c7b12.png">
- <del>Use action link styling not button styling for links that start a service</del>

## Consider
- Removing the `sr-only` "Form name" string in front of the form name. Screen reader users will understand this is a form name as much as sighted users will. Over-engineering accessibility in this instance can cause a more verbose and possibly annoying experience. If labeling this as a form name is important, it should either be visible to _all_ users OR semantically linked using a description list similar to how the name is handled below the `h1`.
<img width="1159" alt="Screen Shot 2021-06-15 at 2 36 16 PM" src="https://user-images.githubusercontent.com/14154792/122105906-48f3aa80-cde7-11eb-80db-9f8ce298966f.png">

<hr/>

## Heading inconsistencies
Headings for forms and links are inconsistent across pages:
- Some have the `a` nested in the `h3`, others are `h3` nested in an `a`
- Some use bitter, others use source sans pro
- Some use `b`, others use `h3`
- Some use the form name, others use the form number

![image](https://user-images.githubusercontent.com/14154792/122107721-61fd5b00-cde9-11eb-8152-3b6e8644c10d.png)
![image](https://user-images.githubusercontent.com/14154792/122107736-64f84b80-cde9-11eb-8375-00f69db9a322.png)

### A11y recommendations
- Nest the `a` inside the `h3` e.g. `<h3><a></a></h3>`
- Use `bitter` for `h1` to `h5` following [VA.gov design system typography guidelines](https://design.va.gov/design/typography#headings)
- Never use `b` as an alternative to a heading as it adds no semantic value to screen reader users, use a `h3` instead
- Use whatever is more recognizable (form name vs. form number) for link headings as screen reader users will have access to a headings and links list to navigate through. AT research will be needed to determine what is more helpful to screen reader users.

<hr/>

## Search results list
Search results are compiled together in one large description `dl` list making it hard to determine where one form ends and another begins on some screen readers. It may be better to split up search results by `ol` and `h3` paired with `dl`.

### Before
```html
<dl> <!-- Upon entering description list, will be announced as "description list, 11 items" which mirrors number of definition titles and descriptions -->
  <!-- Search result 1 -->
  <dt> <!-- Upon navigating into list item (title), will be announced as -->
    <p>Form 10-10ez</p>
    <a>Instructions and Enrollment Application for Health Benefits</a>
  </dt>
  <dd> <!-- description details, the description, definition, or value for the preceding term -->
    <dfn>Form last updated:</dfn> <!-- definition element, the term being defined within the context of a definition phrase or sentence -->
      01-17-2020
  </dd>
  <dd> <!-- description details, the description, definition, or value for the preceding term -->
    <dfn>Related to:</dfn> <!-- definition element, the term being defined within the context of a definition phrase or sentence -->
      Health care
  </dd>
  <dd> <!-- link to 10-10ez tool -->
    <a>Go to online tool for 10-10EZ Instructions and Enrollment Application for Health Benefits</a>
  </dd>
  <dd> <!-- download link to 10-10ez form -->
    <a>Download form 10-10EZ (PDF)</a>
  </dd>

    <!-- Search result 2 -->
  <dt> <!-- description title, a term in a description or definition list -->
    <p>Form 10-10ez</p>
    <a>Instructions and Enrollment Application for Health Benefits</a>
  </dt>
  <dd> <!-- description details, the description, definition, or value for the preceding term -->
    <dfn>Form last updated:</dfn> <!-- definition element, the term being defined within the context of a definition phrase or sentence -->
      01-17-2020
  </dd>
  <dd> <!-- description details, the description, definition, or value for the preceding term -->
    <dfn>Related to:</dfn> <!-- definition element, the term being defined within the context of a definition phrase or sentence -->
      Health care
  </dd>
  <dd> <!-- link to 10-10ez tool -->
    <a>Go to online tool for 10-10EZ Instructions and Enrollment Application for Health Benefits</a>
  </dd>
  <dd> <!-- download link to 10-10ez form -->
    <a>Download form 10-10EZ (PDF)</a>
  </dd>
</dl>
```

### Suggested
```html
<ul role="list"> <!-- Upon entering list, will be announced as "List, 3 items" which mirrors number of search results-->

<li> <!-- Upon navigating into list, will be announced as 1 of [Total number of results] -->
  <p id="unique-id-1">Form 10-10EZ</p> <!-- unique id ties this to the `h3` which screen reader users might jump directly to via the headings list -->
  <h3 aria-describedby="unique-id-1"> <!-- h3 sets clear expectation that this is the start of a new result, also makes it discoverable in headings list -->
    <a>
      Instructions and Enrollment Application for Health Benefits
    </a>
  </h3>
  <dl>
    <dt>Form last updated:</dt>
    <dd>01-17-2020</dd>
    <dt>Related to:</dt>
    <dd>Healthcare</dd>
  </dl>
  <a>Go to online tool</a> <!-- not technically describing the form so much as it is just a link -->
  <a>Download form 10-10EZ (PDF)</a> <!-- not technically describing the form so much as it is just a link -->
</li>

<li> <!-- Upon navigating into list item, will be announced as 2 of [Total number of results] -->
  <p id="unique-id-2">Form 10-10EZ</p> <!-- unique id ties this to the `h3` which screen reader users might jump directly to via the headings list -->
  <h3 aria-describedby="unique-id-2"> <!-- h3 sets clear expectation that this is the start of a new result, also makes it discoverable in headings list -->
    <a>
      Instructions and Enrollment Application for Health Benefits
    </a>
  </h3>
  <dl>
    <dt>Form last updated:</dt>
    <dd>01-17-2020</dd>
    <dt>Related to:</dt>
    <dd>Healthcare</dd>
  </dl>
  <a>Go to online tool</a> <!-- not technically describing the form so much as it is just a link -->
  <a>Download form 10-10EZ (PDF)</a> <!-- not technically describing the form so much as it is just a link -->
</li>

</ul>
```

**Comment by 1Copenut** [comment]
Platform agrees with all feedback!

**Comment by MarciMcGuireGCIO** [comment]
@joshkimux I wonder if you possibly followed the link from the search page to the detail page.  Unfortunately, on Staging, that link takes you to the Prod version of the detail page.  An example of a Staging detail page is https://staging.va.gov/find-forms/about-form-10-10ez/

**Comment by joshkimux** [comment]
@MarciMcGuireGCIO I did! Thank you for the heads up

**Comment by MarciMcGuireGCIO** [comment]
@joshkimux Yay!  It's so easy to do.

**Comment by MarciMcGuireGCIO** [comment]
@joshkimux - I wanted to check and see if you have updated guidance since the original feedback was based on the Production page vs. the one on Staging.   https://staging.va.gov/find-forms/about-form-10-10ez/   Thanks!

**Comment by joshkimux** [comment]
@MarciMcGuireGCIO Looks like all original feedback remains the same! The only difference lies in the heading for related forms and instruction names which are bitter, colored, and underlined 👍

**Comment by MarciMcGuireGCIO** [comment]
Ok, thank you, @joshkimux!  cc @zacharymorel

**Comment by MarciMcGuireGCIO** [comment]
@zacharymorel let's groom this one and possibly fit it into the next sprint.

**Comment by zacharymorel** [comment]
Hey @joshkimux I was scanning this before we go to groom this. 

Can you revist this part of the review and then update it here?
1. That is looking at the current page live on prod. Make sure there is the **staging** subdomain when on a form detail page. https://staging.va.gov/find-forms/about-form-10-10ez/
![Screen Shot 2021-06-30 at 2 37 28 PM](https://user-images.githubusercontent.com/26075258/124013909-e25ab900-d9b0-11eb-82d7-ad1a2f382b60.png)

**Comment by joshkimux** [comment]
@zacharymorel that was there to point out inconsistencies to be thought through for your team's designer 🙇 

Guidance relevant to the semantic markup:
- Nest the a inside the heading e.g. `<h3><a></a></h3>`
- Use font family bitter for h1 to h5 following VA.gov design system typography guidelines
- Use `h3` instead of `b` element for the light blue cards (see below)
<img width="356" alt="Screen Shot 2021-06-30 at 2 56 04 PM" src="https://user-images.githubusercontent.com/14154792/124016129-4c745d80-d9b3-11eb-81d8-34afada9cdf4.png">


Open question for designer/team: 
- Use whatever is more recognizable (form name vs. form number) for link headings as screen reader users will have access to a headings and links list to navigate through. AT research will be needed to determine what is more helpful to screen reader users.

**Comment by zacharymorel** [comment]
Thank you @joshkimux , this clears it up quite a bit.

**Comment by MarciMcGuireGCIO** [comment]
@callen2563 adding you since there is some design-specific feedback, but there are no launch blockers, so we are putting on the back burner to possibly work on in Sprint 52,

**Comment by MarciMcGuireGCIO** [comment]
Nothing here that the feedback related to relying on color is a global element that the Decision Tools team does not manage.  The remaining were captured in other tickets following the Staging review, so I am closing this ticket.
